sarif/operations/diff_op.py

```diff
@@ -98,69 +100,73 @@ def print_diff(
     return ret
 
 
-def _find_new_occurrences(new_records, old_records, issue_code_and_desc):
-    old_occurrences = [
-        r
-        for r in old_records
-        if combine_record_code_and_description(r) == issue_code_and_desc
-    ]
+def _find_new_occurrences(new_records, old_records):
+    old_occurrences = old_records
     new_occurrences_new_locations = []
     new_occurrences_new_lines = []
     for new_record in new_records:
-        if combine_record_code_and_description(new_record) == issue_code_and_desc:
-            (new_location, new_line) = (True, True)
-            for old_record in old_occurrences:
-                if old_record["Location"] == new_record["Location"]:
-                    new_location = False
-                    if old_record["Line"] == new_record["Line"]:
-                        new_line = False
-                        break
-            if new_location:
-                if new_record not in new_occurrences_new_locations:
-                    new_occurrences_new_locations.append(new_record)
-            elif new_line:
-                if new_record not in new_occurrences_new_lines:
-                    new_occurrences_new_lines.append(new_record)
+        (new_location, new_line) = (True, True)
+        for old_record in old_occurrences:
```

Comment:
I think in the worst case it could be O(n^2) where n is the number of occurrences of a specific issue type - in the previous version n was the total number of issues in the SARIF file.  So this code change has improved the situation, I've added a comment about the runtime complexity but I'm keen not to try to optimise it due to the risk of bugs.  If users have thousands of occurrences of an issue, they should be fixing them or turning off the rule, not tracking the changes in the locations!